explorer/client/cypress/e2e/e2e.cy.ts

```diff
@@ -0,0 +1,284 @@
+            cy.get('[data-testid="lastBtn"]').filter(':visible').click();
+            cy.get(nftObject(1)).click();
+            cy.get('#objectID').contains('CollectionObject');
+        });
+
+        it('where last and next disappear in final page', () => {
+            const address = 'ownsAllAddress';
+            cy.visit(`/addresses/${address}`);
+            cy.get('[data-testid="lastBtn"]').filter(':visible').click();
+
+            //Back and First buttons are not disabled:
+            cy.get('[data-testid="backBtn"]')
+                .filter(':visible')
+                .should('be.enabled');
+            cy.get('[data-testid="firstBtn"]')
+                .filter(':visible')
+                .should('be.enabled');
+
+            //Next and Last buttons are disabled:
+            cy.get('[data-testid="nextBtn"]')
+                .filter(':visible')
+                .should('be.disabled');
+            cy.get('[data-testid="lastBtn"]')
+                .filter(':visible')
+                .should('be.disabled');
+        });
+
+        it('to go back a page', () => {
+            const address = 'ownsAllAddress';
+            cy.visit(`/addresses/${address}`);
+
+            cy.get('[data-testid="lastBtn"]').filter(':visible').click();
+            cy.get('[data-testid="backBtn"]').filter(':visible').click();
+            cy.get(nftObject(1)).click();
+            cy.get('#objectID').contains('player5');
+        });
+
+        it('to go to first page', () => {
+            const address = 'ownsAllAddress';
+            cy.visit(`/addresses/${address}`);
+            cy.get('[data-testid="lastBtn"]').filter(':visible').click();
+            cy.get('[data-testid="backBtn"]').filter(':visible').click();
+            cy.get('[data-testid="firstBtn"]').filter(':visible').click();
+            cy.get(nftObject(1)).click();
+            cy.get('#objectID').contains('ChildObjectWBrokenImage');
+        });
+
+        it('where first and back disappear in first page', () => {
+            const address = 'ownsAllAddress';
+            cy.visit(`/addresses/${address}`);
+
+            //Back and First buttons are disabled:
+            cy.get('[data-testid="backBtn"]')
+                .filter(':visible')
+                .should('be.disabled');
+            cy.get('[data-testid="firstBtn"]')
+                .filter(':visible')
+                .should('be.disabled');
+
+            //Next and Last buttons are not disabled:
+            cy.get('[data-testid="nextBtn"]')
+                .filter(':visible')
+                .should('be.enabled');
+            cy.get('[data-testid="lastBtn"]')
+                .filter(':visible')
+                .should('be.enabled');
+        });
+    });
+
+    describe('Group View', () => {
+        it('evaluates balance', () => {
+            const address = 'ownsAllAddress';
+            cy.visit(`/addresses/${address}`);
+
+            cy.get('#groupCollection').contains('0x2::USD::USD');
```

Comment:
We should be more specific in checking that the correct type is placed next to the correct balance. Under the original tests we had a `coinGroup` function that could help with this.

Reply:
Yeah I can tweak these assertions to bring back the more structural testing.